### PR TITLE
test(api): per-pillar handle smoke harness (#2913)

### PR DESCRIPTION
**Pull request by knoxio**
## Summary

Adds an integration smoke harness that exercises every query procedure mounted on each pillar's tRPC router against a fresh per-pillar SQLite opened through the canonical `open<Pillar>Db(':memory:')` path. Any procedure that throws `SqliteError: no such table` fails the harness — the missing-from-journal table surfaces in CI instead of prod boot.

Catches the Track N4 (#2908) regression class. Covers **finance, core, media, inventory, cerebrum** — the pillars with cutover slices. Food + lists are deferred until Track J + K cutovers complete (no `__integration__/` dirs created for them).

## What's in this PR

- `apps/pops-api/src/shared/pillar-smoke-harness.ts` — generic reflection-based harness. Enumerates `<pillar>.*` queries from `appRouter._def.procedures`, walks the caller proxy, races each call against a per-procedure timeout, and reports only `no such table` failures. Non-table errors (Zod validation, NotFound, FK violations on empty seed) are tolerated by design — they prove the SQL parser reached the table.
- `apps/pops-api/src/shared/pillar-smoke-harness.test.ts` — 11 unit tests against a hand-rolled fake router/caller (so the harness can't quietly miss its own regression).
- `apps/pops-api/src/modules/<pillar>/__integration__/<pillar>-handle-coverage.test.ts` — one file per pillar (finance, core, media, inventory, cerebrum + ego sibling). Each opens a fresh per-pillar `:memory:` DB, injects via `set<Pillar>Db`, and runs the harness.

## Adversarial verification

Manually mutated two migrations to confirm the harness catches the bug class:

1. **`wish_list` removed from `packages/finance-db/migrations/0053_finance_pillar_baseline.sql`** → finance harness fails with:
   - `finance.wishlist.list: no such table: wish_list`
   - `finance.wishlist.get: no such table: wish_list`

2. **`tag_vocabulary` removed from `packages/finance-db/migrations/0026_little_frank_castle.sql`** → core harness fails with:
   - `core.tagRules.listVocabulary: no such table: tag_vocabulary` (cross-pillar — proc lives on `core.*` but reads via `getFinanceDrizzle()`, exactly the N4 #2908 shape)

Both restored before commit.

## Performance

Per-pillar runtime on a warm vitest worker:
- finance: ~250ms
- inventory: ~250ms
- media: ~700ms (timeouts on Plex procs)
- cerebrum: ~600ms (timeouts on retrieval procs that try Redis/embedding API)
- core: ~1.5s (timeouts on BullMQ jobs procs that try Redis)

Total across all five pillars: ~4.4s. Well inside the <5s/pillar budget.

## Hard rules

- No `as any`, no `as unknown as Type`, no `ts-ignore`, no `eslint-disable`.
- No `setTimeout` masks, no `vi.useFakeTimers`. Real timers, real I/O, in-memory DBs. The harness's per-procedure 250ms guard is hygiene against external-dependency hangs (BullMQ → Redis), not a flakiness mask — a procedure that genuinely needs >250ms of SQL on `:memory:` is broken on its own merits.
- `pnpm -F pops-api run typecheck` ✅
- `pnpm lint` ✅
- `pnpm -F pops-api run test -- __integration__` ✅ (5 files, 15 tests)
- `pnpm -F pops-api run test` ✅ (312 files, 4902 tests)

## Test plan

- [x] Harness unit tests cover: enumeration, dispatch, ignorePaths, inputs map, missing-table detection (top-level + err.cause), non-table errors not counted, per-procedure timeout.
- [x] Adversarial mutation of `wish_list` CREATE catches in finance harness.
- [x] Adversarial mutation of `tag_vocabulary` CREATE catches in core harness (cross-pillar via `getFinanceDrizzle()`).
- [x] All existing tests in pops-api still pass.

Closes #2913

### CODE REVIEW

File: apps/pops-api/src/shared/pillar-smoke-harness.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
+/**
+ * Per-pillar tRPC handle smoke harness.
+ *
+ * Why: each pillar (finance, core, media, inventory, cerebrum) now owns
+ * its own SQLite file opened by a per-pillar `openFooDb()`. When a
+ * procedure is cut over to a per-pillar `getFooDrizzle()` handle but
```
[the remark this answers is not in the document]
Comment (knoxio) [reply apps/pops-api/src/shared/pillar-smoke-harness.ts:6]: Food + lists deferral is intentional and called out explicitly in the issue body: "For food + lists: their migration tracks (J + K) aren't complete yet; document in a comment that the smoke harness should be added once those land, but DON'T add the test files now (they'd be empty placeholders)." The deferral is documented in the PR body. Issue #2913 acceptance criteria match the PR scope.
